### PR TITLE
Refactors PublisherToggle into redux component

### LINKED ISSUE #9323: Redux component - PublisherToggle
## Test plan

https://github.com/brave/browser-laptop/pull/9329#issue-234370022

----

**Describe the issue you encountered:**
Refactor `PublisherToggle` into redux component

### PR DESCRIPTION
**Pull request by NejcZdovc**
Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request [as needed](https://github.com/brave/browser-laptop/wiki/Pull-request-process).

Resolves #9323

Auditors: @bridiver @bsclifton

Test Plan:
- toggle should be hidden when payment is disabled
- toggle should be hidden when you are on about page
- toggle should be hidden when you are on non http/https site

Reviewer Checklist:

Tests

- [ ] Adequate test coverage exists to prevent regressions
- [ ] Tests should be independent and work correctly when run individually or as a suite [ref](https://github.com/brave/browser-laptop/wiki/Code-Guidelines#test-dependencies)
- [ ] New files have MPL2 license header

### CODE REVIEW

File: app/common/state/publisherState.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const Immutable = require('immutable')
```
Comment (bridiver) [inline app/common/state/publisherState.js]: not sure if this file makes sense as a state helper. The state helpers should own a portion of the state and this is more of a set of utility functions that access other state helpers

Comment (NejcZdovc) [inline app/common/state/publisherState.js]: it's not done yet :smiley: both functions used here get something from state

Comment (bridiver) [inline app/common/state/publisherState.js]: yes, they get something from the state, but they don't "own" part of the state. I think maybe better as util functions?

Comment (NejcZdovc) [inline app/common/state/publisherState.js]: sure I can move it into `publisherUtils.js`

Comment (NejcZdovc) [inline app/common/state/publisherState.js]: done

**Comment by NejcZdovc** [comment]
Let's merge this after #9415 is merged. This way unit test should stop failing.

**Comment by NejcZdovc** [comment]
PR rebased and ready for a review

**Comment by NejcZdovc** [comment]
Unit test is failing, let me check out what is going on

**Comment by NejcZdovc** [comment]
Unit test fixed

File: app/renderer/components/navigation/urlBarIcon.js [hunk 2]
```diff
@@ -115,6 +115,7 @@ class UrlBarIcon extends React.Component {
   }
 
   mergeProps (state, ownProps) {
+    console.log(state.get('tabs').toJS())
```
Comment (bridiver) [inline app/renderer/components/navigation/urlBarIcon.js]: ?

Comment (NejcZdovc) [inline app/renderer/components/navigation/urlBarIcon.js]: Removed

**Pull request merged**